dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <xtensa/xtensa.dtsi>
+#include <mem.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <0>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+
+		cpu1: cpu@1 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <1>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+
+		cpu2: cpu@2 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <2>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+	};
+
+	power-states {
+		idle: idle {
+			compatible = "zephyr,power-state";
+			power-state-name = "runtime-idle";
+			min-residency-us = <0>;
+			exit-latency-us = <0>;
+		};
+		suspend: suspend {
+			compatible = "zephyr,power-state";
+			power-state-name = "suspend-to-idle";
+			min-residency-us = <200>;
+			exit-latency-us = <100>;
+		};
+		/* PM_STATE_SOFT_OFF can be entered only by calling pm_state_force.
+		 * The procedure is triggered by IPC from the HOST (SET_DX).
+		 */
+		off: off {
+			compatible = "zephyr,power-state";
+			power-state-name = "soft-off";
+			min-residency-us = <2147483647>;
```

Comment:
A min residency of INT_MAX is more than half an hour.  Not sure where this comes from, and I'll buy that this is probably ignored by existing code, but that actual number is almost certainly a lie.  This doesn't seem to be a pattern used anywhere else in Zephyr, and I don't see any code trying to interpret INT_MAX specially here.  Can we put a real number in here that matches the design intent of the field?

Reply:
@andyross the reason why this state has such a strange value is explained in the comments above the definition. FW will enter this state only on the request from the SW. There is of course permanent lock on application side but I wanted to be sure that this state will not be entered by mistake.

During the first review, it was proposed to remove this state from dts, since it can be entered only by using **_pm_state_force_**. But in my opinion, it would be unfair for a person who will come into contact with this code in some time and see that the processor enters the state that according to board definition is not even supported.

Do you have a suggestion what would be better than the maximum value?

Reply:
If the PM scheduling code tries to put you in a state you can't support, don't we have an API for that already by returning the actual state or an error code or something?  I mean, at the end of the day the PM integration is clearly empowered to implement rules like "don't power the core off unless we've received a power-down message from the host" regardless of what the PM layer wants.  So I don't see why we need to lie to the PM layer, I guess.

(For clarity: this comment isn't driving my -1.  It just seems clumsy to me and I suspect there's a misunderstanding of the PM architecture behind it, though I'm not the expert to make that call.)

Reply:
@andyross What value would not be a lie, since there is no number that apply to this state? And to be clear if you misunderstood me before. This is valid state that is supported by each core and will be entered by them.

Our custom pm policy will guarantee that this state will not be entered any other way than on host request. The only other solution I see is to set min-residency-us equal 0 and remove this state from cpu-power-states list (or maybe just set it to 0).